clients/flash/air-client/src/org/bigbluebutton/air/screenshare/views/ScreenshareDock.as

```diff
@@ -0,0 +1,224 @@
+package org.bigbluebutton.air.screenshare.views {
+	import flash.events.AsyncErrorEvent;
+	import flash.events.NetStatusEvent;
+	import flash.events.StageVideoAvailabilityEvent;
+	import flash.events.StageVideoEvent;
+	import flash.geom.Rectangle;
+	import flash.media.StageVideo;
+	import flash.media.StageVideoAvailability;
+	import flash.net.NetConnection;
+	import flash.net.NetStream;
+	
+	import spark.components.Group;
+	
+	public class ScreenshareDock extends Group {
+		private var _ns:NetStream;
+		
+		private var _conn:NetConnection;		
+		private var _streamId:String;	
+		private var _origVidWidth:int;
+		private var _origVidHeight:int;
+		private var _sv:StageVideo;
+		private var _ssView:ScreenshareView;
+		private var _usingStageVideo:Boolean = false;
+		private var _usingVideo:Boolean = false;
+		private var _played:Boolean = false;
+		private var _screenshareRunningListener:Function;
+		
+		public function ScreenshareDock():void {	}
+		
+		private function onStageVideoState(event:StageVideoAvailabilityEvent):void {
+			var available:Boolean = (event.availability == StageVideoAvailability.AVAILABLE);
+			
+			//available = false; // for testing to force using Video instead of StageVideo!!!
+			
+			//trace("************ ScreenshareView: STAGE VIDEO available=" + available);
+			stage.removeEventListener(StageVideoAvailabilityEvent.STAGE_VIDEO_AVAILABILITY, onStageVideoState);
+			// Detect if StageVideo is available and decide what to do in toggleStageVideo 
+			toggleStageVideo(available);
+		}
+		
+		private function toggleStageVideo(on:Boolean):void {
+			_ns = new NetStream(_conn);
+			_ns.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
+			_ns.addEventListener(AsyncErrorEvent.ASYNC_ERROR, onAsyncError);
+			_ns.client = this;
+			_ns.bufferTime = 0;
+			_ns.receiveVideo(true);
+			_ns.receiveAudio(false);
+			
+			//trace("**** Container w=" + this.width + ",h=" + this.height + " video w=" + _origVidWidth + ",h=" + _origVidHeight);
+			
+			var viewPort:Rectangle = positionAndSize(this.width, this.height, _origVidWidth, _origVidHeight);
+			
+			// To choose StageVideo attach the NetStream to StageVideo 
+			if (on) {				
+				_usingStageVideo = true;
+				if (_sv == null) {
+					//trace("***** Using StageVideo length=" + stage.stageVideos.length);
+					_sv = stage.stageVideos[0];
+					_sv.viewPort = viewPort;

```

Comment:
The viewport will be using local x/y coordinates, but the StageVideo will be expecting global coordinates. The properties should be transformed with localToGlobal before attaching to the viewPort.